Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -2769,12 +2769,15 @@ static inline bool rectIsTooBigForSelection(const IntRect& blockRect, const Loca
         auto position = frame->selection().selection().visibleStart();
         range = wordRangeFromPosition(position);
         textForRange = plainTextForContext(range);
-        
         // If 'originalText' is not the same as 'textForRange' we need to move 'range'
         // forward such that it matches the original selection as much as possible.
         if (foldQuoteMarks(textForRange) != originalTextWithFoldedQuoteMarks) {
             // Search for the original text near the selection caret.
-            if (auto searchRange = rangeExpandedAroundPositionByCharacters(position, numGraphemeClusters(originalText))) {
+            auto characterCount = numGraphemeClusters(originalText);
```

Comment:
unrelated, but when should `numGraphemeClusters` be used instead of `.length()` on a string?

Reply:
The `length()` of the string only counts the number of UTF16 characters, which is different from grapheme cluster count (the latter of which is much closer to what we'd regard as a single selectable/visible unit).

For example — a string containing just "👋🏻" has a length of 4, but represents a single grapheme cluster (you shouldn't be able to move the caret in between the hand waving codepoint and the skin color codepoint).

Thanks for the review!